cypress-io/cypress
# fix: force Webpack to emit assets on error

- Closes #23219

### User facing changelog


Compile errors will now be surfaced in the command log during testing for Angular and Next projects

### Additional details


Some frameworks (Angular and Next) configure Webpack to not emit assets that have errored. So, if there's a compile error during development in these projects, instead of serving the module that has the compilation error and throwing it, Webpack serves the successful asset from the previous build. The result is that the compilation error only appears in the console, and our command log and AUT look fine - as if nothing has happened.

This behavior is undesirable because if there is a compilation error, we want to fail the test and make the user aware. This PR adds `emitOnErrors: true` (`noEmitOnErrors: false` for Webpack 4) to our default Webpack configuration to override these values and force Webpack to emit modules even when there is an error regardless of the framework configuration.

### Steps to test


1. Revert my changes in `makeDefaultWebpackConfig.ts`
2. Run the tests for `angular.cy.ts` and `next.cy.ts` in `npm/webpack-dev-server`
3. Verify that the tests fail because errors do not appear in the command log when a compilation error is written to the component file
4. Put the changes to `makeDefaultWebpackConfig.ts` back and repeat step 2
5. Verify that the tests pass now and the errors appear in the command log as expected

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
